--- a/ext/x264/gstx264enc.c	2016-11-04 17:07:16.000000000 +0000
+++ b/ext/x264/gstx264enc.c	2018-10-23 15:16:06.803205824 +0000
@@ -557,7 +557,7 @@
 {
   GValue fmt = G_VALUE_INIT;
 
-  if (x264_bit_depth == 8) {
+  if (X264_BIT_DEPTH == 8) {
     GST_INFO ("This x264 build supports 8-bit depth");
     if (x264_chroma_format_local == 0) {
       set_value (&fmt, 5, "I420", "YV12", "Y42B", "Y444", "NV12");
@@ -570,7 +570,7 @@
     } else {
       GST_ERROR ("Unsupported chroma format %d", x264_chroma_format_local);
     }
-  } else if (x264_bit_depth == 10) {
+  } else if (X264_BIT_DEPTH == 10) {
     GST_INFO ("This x264 build supports 10-bit depth");
 
     if (G_BYTE_ORDER == G_LITTLE_ENDIAN) {
@@ -600,7 +600,7 @@
     }
   } else {
     GST_ERROR ("Unsupported bit depth %d, we only support 8-bit and 10-bit",
-        x264_bit_depth);
+        X264_BIT_DEPTH);
   }
 
   if (G_VALUE_TYPE (&fmt) != G_TYPE_INVALID)
