For now, allow to build with older meson


--- cairo-1.18.2/meson.build.~2~	2024-11-08 16:57:03.569405406 +0000
+++ cairo-1.18.2/meson.build	2024-11-08 18:38:22.524392951 +0000
@@ -1,7 +1,7 @@
 project('cairo', 'c',
-  meson_version: '>= 1.3.0',
+  meson_version: '>= 1.2.2',
   version: run_command(find_program('version.py'), check: true).stdout().strip(),
-  default_options: ['c_std=gnu11,c11',
+  default_options: ['c_std=gnu11',
                     'warning_level=2'],
 )
 
@@ -332,9 +332,6 @@
         conf.set('CAIRO_CAN_TEST_TTX_FONT', 1)
       endif
     endif
-    if cc.has_define('FT_LOAD_NO_SVG', dependencies: freetype_dep, prefix: '#include <freetype/freetype.h>')
-      conf.set('HAVE_FT_LOAD_NO_SVG', 1)
-    endif
     if freetype_dep.version().version_compare(freetype_colrv1_required_version) and \
        cc.has_function('FT_Get_Color_Glyph_Paint', dependencies: freetype_dep)
       conf.set('HAVE_FT_COLR_V1', 1)
