--- a/configure	2017-03-19 20:30:27.000000000 +0000
+++ b/configure	2018-11-23 19:01:25.155787824 +0000
@@ -15714,7 +15714,7 @@
     if test -d /usr/share/zoneinfo; then
 	ZONEINFO_DIR="/usr/share/zoneinfo"
     elif test -d /usr/share/lib/zoneinfo/tab; then
-	ZONEINFO_DIR="/usr/share/lib/zoneinfo/tab"
+	ZONEINFO_DIR="/usr/share/lib/zoneinfo"
     else
 	as_fn_error $? "Can't find zoneinfo directory. Use --with-zoneinfo-dir" "$LINENO" 5
     fi
