--- samba-4.11.6/source4/heimdal/lib/hcrypto/libtommath/tommath.h~	2019-12-06 09:46:56.000000000 +0000
+++ samba-4.11.6/source4/heimdal/lib/hcrypto/libtommath/tommath.h	2020-09-25 14:52:59.343508130 +0000
@@ -31,6 +31,11 @@
    #define MAX(x,y) ((x)>(y)?(x):(y))
 #endif
 
+#if defined(__sun) && defined(ERR)
+// ERR conflicts with Solaris regset.h
+#undef ERR
+#endif
+
 #ifdef __cplusplus
 extern "C" {
 
