--- rustc-1.60.0-src/src/llvm-project/llvm/CMakeLists.txt~	2022-05-02 17:38:27.424917237 +0000
+++ rustc-1.60.0-src/src/llvm-project/llvm/CMakeLists.txt	2022-05-02 17:38:47.033561745 +0000
@@ -615,7 +615,7 @@
 
 option(LLVM_BUILD_BENCHMARKS "Add LLVM benchmark targets to the list of default
 targets. If OFF, benchmarks still could be built using Benchmarks target." OFF)
-option(LLVM_INCLUDE_BENCHMARKS "Generate benchmark targets. If OFF, benchmarks can't be built." ON)
+option(LLVM_INCLUDE_BENCHMARKS "Generate benchmark targets. If OFF, benchmarks can't be built." OFF)
 
 option (LLVM_BUILD_DOCS "Build the llvm documentation." OFF)
 option (LLVM_INCLUDE_DOCS "Generate build targets for llvm documentation." ON)
