--- rustc-1.50.0-src/library/std/src/sys/pal/unix/mod.rs	2021-02-10 17:36:44.000000000 +0000
+++ rustc-1.50.0-src/library/std/src/sys/pal/unix/mod.rs	2021-03-05 15:47:24.847514661 +0000
@@ -259,6 +259,7 @@
         #[link(name = "posix4")]
         #[link(name = "pthread")]
         #[link(name = "resolv")]
+        #[link(name = "nsl")]
         extern "C" {}
     } else if #[cfg(target_os = "illumos")] {
         #[link(name = "socket")]
--- rustc-1.66.0-src/compiler/rustc_target/src/spec/base/solaris.rs	2022-12-12 16:02:12.000000000 +0000
+++ rustc-1.66.0-src/compiler/rustc_target/src/spec/base/solaris.rs	2022-12-15 19:02:18.783814001 +0000
@@ -1,6 +1,23 @@
-use crate::spec::{cvs, Cc, LinkerFlavor, TargetOptions};
+use crate::spec::{LinkArgs, FramePointer, cvs, Cc, LinkerFlavor, TargetOptions};
 
 pub fn opts() -> TargetOptions {
+    let mut late_link_args = LinkArgs::new();
+    late_link_args.insert(
+        LinkerFlavor::Unix(Cc::Yes),
+        vec![
+            // LLVM will insert calls to the stack protector functions
+            // "__stack_chk_fail" and "__stack_chk_guard" into code in native
+            // object files.  Some platforms include these symbols directly in
+            // libc, but at least historically these have been provided in
+            // libssp.so on illumos and Solaris systems.
+            "-lssp".into(),
+
+            // Disable .SUNW_ldynsym sections on Solaris,
+            // as llvm cannot read the linked SHT_SYMTAB_SHNDX
+            "-Wl,-znoldynsym".into(),
+        ],
+    );
+
     TargetOptions {
         os: "solaris".into(),
         dynamic_linking: true,
@@ -10,6 +27,8 @@
         linker_flavor: LinkerFlavor::Unix(Cc::Yes),
         limit_rdylib_exports: false, // Linker doesn't support this
         eh_frame_header: false,
+        frame_pointer: FramePointer::Always,
+        late_link_args,
 
         ..Default::default()
     }
--- rustc-1.78.0-src/src/bootstrap/src/core/builder.rs.~1~	2024-05-10 19:48:28.821487288 +0000
+++ rustc-1.78.0-src/src/bootstrap/src/core/builder.rs	2024-05-10 19:50:54.674454903 +0000
@@ -2415,16 +2415,20 @@
                 self.rustflags.arg("-Zosx-rpath-install-name");
                 Some(format!("-Wl,-rpath,@loader_path/../{libdir}"))
             } else if !target.is_windows() && !target.contains("aix") && !target.contains("xous") {
-                self.rustflags.arg("-Clink-args=-Wl,-z,origin");
+                self.rustflags.arg("-Clink-arg=-Wl,-z,origin");
                 Some(format!("-Wl,-rpath,$ORIGIN/../{libdir}"))
             } else {
                 None
             };
             if let Some(rpath) = rpath {
-                self.rustflags.arg(&format!("-Clink-args={rpath}"));
+                self.rustflags.arg(&format!("-Clink-arg={rpath}"));
             }
         }
 
+        // Disable .SUNW_ldynsym sections on Solaris,
+        // as llvm cannot read the linked SHT_SYMTAB_SHNDX
+        self.rustflags.arg("-Clink-arg=-Wl,-znoldynsym");
+
         for arg in linker_args(builder, compiler.host, LldThreads::Yes) {
             self.hostflags.arg(&arg);
         }
