--- rustc-1.86.0-src/library/std/src/sys/pal/unix/mod.rs.~1~	2025-04-03 11:06:13.094850783 +0000
+++ rustc-1.86.0-src/library/std/src/sys/pal/unix/mod.rs	2025-04-03 11:07:36.928278011 +0000
@@ -390,6 +390,7 @@
         #[link(name = "posix4")]
         #[link(name = "pthread")]
         #[link(name = "resolv")]
+        #[link(name = "nsl")]
         unsafe extern "C" {}
     } else if #[cfg(target_os = "illumos")] {
         #[link(name = "socket")]
--- rustc-1.83.0-src/compiler/rustc_target/src/spec/base/solaris.rs.~1~	2024-11-26 10:12:09.000000000 +0000
+++ rustc-1.83.0-src/compiler/rustc_target/src/spec/base/solaris.rs	2024-11-28 17:24:14.405334800 +0000
@@ -1,6 +1,23 @@
-use crate::spec::{Cc, LinkerFlavor, TargetOptions, cvs};
+use crate::spec::{LinkArgs, FramePointer, Cc, LinkerFlavor, TargetOptions, cvs};
 
 pub(crate) fn opts() -> TargetOptions {
+    let mut late_link_args = LinkArgs::new();
+    late_link_args.insert(
+        LinkerFlavor::Unix(Cc::Yes),
+        vec![
+            // LLVM will insert calls to the stack protector functions
+            // "__stack_chk_fail" and "__stack_chk_guard" into code in native
+            // object files.  Some platforms include these symbols directly in
+            // libc, but at least historically these have been provided in
+            // libssp.so on illumos and Solaris systems.
+            "-lssp".into(),
+
+            // Disable .SUNW_ldynsym sections on Solaris,
+            // as llvm cannot read the linked SHT_SYMTAB_SHNDX
+            "-Wl,-znoldynsym".into(),
+        ],
+    );
+
     TargetOptions {
         os: "solaris".into(),
         dynamic_linking: true,
@@ -10,6 +27,8 @@
         linker_flavor: LinkerFlavor::Unix(Cc::Yes),
         limit_rdylib_exports: false, // Linker doesn't support this
         eh_frame_header: false,
+        frame_pointer: FramePointer::Always,
+        late_link_args,
 
         ..Default::default()
     }
--- rustc-1.88.0-src/src/bootstrap/src/core/builder/cargo.rs.~1~	2025-06-27 19:19:10.424422987 +0000
+++ rustc-1.88.0-src/src/bootstrap/src/core/builder/cargo.rs	2025-06-27 19:21:31.983586877 +0000
@@ -251,16 +251,20 @@
                 && !target.contains("aix")
                 && !target.contains("xous")
             {
-                self.rustflags.arg("-Clink-args=-Wl,-z,origin");
+                self.rustflags.arg("-Clink-arg=-Wl,-z,origin");
                 Some(format!("-Wl,-rpath,$ORIGIN/../{libdir}"))
             } else {
                 None
             };
             if let Some(rpath) = rpath {
-                self.rustflags.arg(&format!("-Clink-args={rpath}"));
+                self.rustflags.arg(&format!("-Clink-arg={rpath}"));
             }
         }
 
+        // Disable .SUNW_ldynsym sections on Solaris,
+        // as llvm cannot read the linked SHT_SYMTAB_SHNDX
+        self.rustflags.arg("-Clink-arg=-Wl,-znoldynsym");
+
         for arg in linker_args(builder, compiler.host, LldThreads::Yes) {
             self.hostflags.arg(&arg);
         }
