--- a/goo/gstrtod.cc	2010-06-08 21:06:31.000000000 +0000
+++ b/goo/gstrtod.cc	2018-09-03 13:16:44.666065327 +0000
@@ -27,6 +27,10 @@
 #include <cstdlib>
 #include <cstring>
 
+#include <string.h>
+#include <locale.h>
+#include <stdlib.h>
+
 #define ascii_isspace(c) \
   (c == ' ' || c == '\f' || c == '\n' || c == '\r' || c == '\t' || c == '\v')
 #define ascii_isdigit(c) \
--- a/glib/demo/main.c	2010-11-02 19:20:36.000000000 +0000
+++ b/glib/demo/main.c	2018-09-03 13:41:37.904967322 +0000
@@ -158,7 +158,7 @@
 		return 1;
 	}
 
-	if (!g_thread_supported ())
+	/*if (!g_thread_supported ())*/
 		g_thread_init (NULL);
 
 	gtk_init (&argc, &argv);
