--- firefox-153.1.0/intl/icu/source/i18n/measunit_extra.cpp.~1~	2026-08-12 00:17:53.000000000 +0000
+++ firefox-153.1.0/intl/icu/source/i18n/measunit_extra.cpp	2026-08-20 19:42:15.044402372 +0000
@@ -37,6 +37,7 @@
 #include "util.h"
 #include <limits.h>
 #include <cstdlib>
+#include <math.h>
 U_NAMESPACE_BEGIN
 
 
@@ -694,7 +695,7 @@
         // Check if the value is integer.
         uint64_t int_result = static_cast<uint64_t>(double_result);
         const double kTolerance = 1e-9;
-        if (abs(double_result - int_result) > kTolerance) {
+        if (fabs(double_result - int_result) > kTolerance) {
             status = kUnitIdentifierSyntaxError;
             return 0;
         }
