--- gnome-screenshot-3.38.0/data/meson.build~	2020-09-10 13:54:34.000000000 +0000
+++ gnome-screenshot-3.38.0/data/meson.build	2021-01-06 13:23:53.591258341 +0000
@@ -22,14 +22,14 @@
   )
 endif
 
-metainfo_file = i18n.merge_file(
-  'appdata',
-  input: 'org.gnome.Screenshot.metainfo.xml.in',
-  output: 'org.gnome.Screenshot.metainfo.xml',
-  po_dir: join_paths(meson.current_source_dir(), '../po'),
-  install: true,
-  install_dir: gnome_screenshot_appdatadir
-)
+#metainfo_file = i18n.merge_file(
+#  'appdata',
+#  input: 'org.gnome.Screenshot.metainfo.xml.in',
+#  output: 'org.gnome.Screenshot.metainfo.xml',
+#  po_dir: join_paths(meson.current_source_dir(), '../po'),
+#  install: true,
+#  install_dir: gnome_screenshot_appdatadir
+#)
 
 # Validate metainfo
 appstream_util = find_program('appstream-util', required: false)
@@ -40,7 +40,7 @@
     args: [
       'validate',
       '--nonet',
-      metainfo_file.full_path()
+#      metainfo_file.full_path()
     ]
   )
 endif
