--- gnome-screenshot-3.38.0/src/screenshot-backend-x11.c~	2020-09-10 13:54:34.000000000 +0000
+++ gnome-screenshot-3.38.0/src/screenshot-backend-x11.c	2021-01-06 15:27:19.873305865 +0000
@@ -23,6 +23,8 @@
 
 #ifdef HAVE_X11
 
+#include <X11/Xatom.h>
+
 #include "screenshot-backend-x11.h"
 
 #include "screenshot-config.h"
@@ -228,6 +230,67 @@
   cairo_region_destroy (invisible_region);
 }
 
+static gboolean
+get_gtk_frame_extents (GdkWindow *window,
+                       GtkBorder *extents)
+{
+  Display *xdisplay;
+  Window xwindow;
+  Atom returned_type;
+  int returned_format;
+  gulong n_items, bytes_after;
+  guchar *property;
+
+  returned_type = None;
+  returned_format = 0;
+  n_items = bytes_after = 0;
+
+  xwindow = GDK_WINDOW_XID (window);
+  xdisplay = GDK_DISPLAY_XDISPLAY (gdk_display_get_default ());
+
+  gdk_error_trap_push ();
+
+  if (XGetWindowProperty (xdisplay, xwindow,
+                          gdk_x11_get_xatom_by_name ("_GTK_FRAME_EXTENTS"),
+                          0, G_MAXLONG, False, XA_CARDINAL,
+                          &returned_type,
+                          &returned_format,
+                          &n_items,
+                          &bytes_after,
+                          &property) != Success || returned_type == None)
+    {
+      if (property)
+        {
+          XFree (property);
+        }
+
+      gdk_error_trap_pop_ignored ();
+
+      return FALSE;
+    }
+
+  if (gdk_error_trap_pop () != Success || returned_format != 32 || returned_type != XA_CARDINAL || n_items != 4)
+    {
+      if (property)
+        {
+          XFree (property);
+        }
+
+      return FALSE;
+    }
+
+  gulong *borders = (gulong *) property;
+
+  extents->left   = (int)borders[0];
+  extents->right  = (int)borders[1];
+  extents->top    = (int)borders[2];
+  extents->bottom = (int)borders[3];
+
+  XFree (property);
+
+  return TRUE;
+}
+
 static void
 screenshot_fallback_get_window_rect_coords (GdkWindow    *window,
                                             GdkRectangle *real_coordinates_out,
@@ -236,9 +299,18 @@
   gint x_orig, y_orig;
   gint width, height;
   GdkRectangle real_coordinates;
+  GtkBorder extents;
 
   gdk_window_get_frame_extents (window, &real_coordinates);
 
+  if (get_gtk_frame_extents (window, &extents))
+    {
+      real_coordinates.x += extents.left / 2;
+      real_coordinates.y += extents.top / 2;
+      real_coordinates.width -= (extents.left + extents.right) / 2;
+      real_coordinates.height -= (extents.top + extents.bottom) / 2;
+    }
+
   x_orig = real_coordinates.x;
   y_orig = real_coordinates.y;
   width  = real_coordinates.width;
@@ -439,6 +511,14 @@
               rec_width = rectangles[i].width / scale_factor - (frame_offset.left + frame_offset.right);
               rec_height = rectangles[i].height / scale_factor - (frame_offset.top + frame_offset.bottom);
 
+              GtkBorder extents;
+
+              if (get_gtk_frame_extents (window, &extents))
+                {
+                  rec_width -= (extents.left + extents.right) / 2;
+                  rec_height -= (extents.top + extents.bottom) / 2;
+                }
+
               if (real_coords.x < 0)
                 {
                   rec_x += real_coords.x;
