Should investigate and send upstream.

--- a/comm/ldap/c-sdk/libraries/libldap/tmplout.c
+++ b/comm/ldap/c-sdk/libraries/libldap/tmplout.c
@@ -48,11 +48,6 @@
 #  include <time.h> /* for struct tm and ctime */
 #endif
 
-/* This is totally lame, since it should be coming from time.h, but isn't. */
-#if defined(SOLARIS)
-char* ctime_r(const time_t*, char*, int);
-#endif
-
 static int do_entry2text(LDAP* ld, char* buf, char* base, LDAPMessage* entry,
                          struct ldap_disptmpl* tmpl, char** defattrs,
                          char*** defvals, writeptype writeproc, void* writeparm,
--- a/comm/ldap/c-sdk/libraries/libldap/os-ip.c
+++ b/comm/ldap/c-sdk/libraries/libldap/os-ip.c
@@ -64,6 +64,8 @@
 /* XXX value for macintosh (if applicable)? */
 #endif
 
+#include <sys/filio.h>
+
 #include "ldap-int.h"
 #ifdef NSLDAPI_CONNECT_MUST_NOT_BE_INTERRUPTED
 #  include <signal.h>
