--- a/lisp/term/xterm.el	2021-05-06 20:18:24.573271545 +0000
+++ b/lisp/term/xterm.el	2021-05-06 22:43:29.467971518 +0000
@@ -877,7 +877,7 @@
     (xterm--init-frame-title))
   ;; Unconditionally enable bracketed paste mode: terminals that don't
   ;; support it just ignore the sequence.
-  (xterm--init-bracketed-paste-mode)
+  ;; (xterm--init-bracketed-paste-mode)
   ;; We likewise unconditionally enable support for focus tracking.
   (xterm--init-focus-tracking)
 
