--- a/js/src/util/NativeStack.cpp	2018-10-17 21:39:31.000000000 +0000
+++ b/js/src/util/NativeStack.cpp	2018-12-07 15:41:34.541880819 +0000
@@ -32,7 +32,7 @@
     return static_cast<void*>(pTib->StackBase);
 }
 
-#elif defined(SOLARIS)
+#elif defined(__sun__)
 
 #include <ucontext.h>
 
--- a/ipc/chromium/src/base/time_posix.cc	2018-10-17 21:39:28.000000000 +0000
+++ b/ipc/chromium/src/base/time_posix.cc	2018-12-09 22:21:52.733110689 +0000
@@ -24,6 +24,40 @@
 #include "base/basictypes.h"
 #include "base/logging.h"
 
+#ifdef __sun__
+
+extern "C" {
+  time_t timegm(struct tm *t)
+  {
+    time_t tl, tb;
+    struct tm *tg;
+
+    tl = mktime (t);
+    if (tl == -1)
+      {
+        t->tm_hour--;
+        tl = mktime (t);
+        if (tl == -1)
+          return -1; /* can't deal with output from strptime */
+        tl += 3600;
+      }
+    tg = gmtime (&tl);
+    tg->tm_isdst = 0;
+    tb = mktime (tg);
+    if (tb == -1)
+      {
+        tg->tm_hour--;
+        tb = mktime (tg);
+        if (tb == -1)
+          return -1; /* can't deal with output from gmtime */
+        tb += 3600;
+      }
+    return (tl - (tb - tl));
+  }
+}
+
+#endif
+
 namespace base {
 
 // The Time routines in this file use standard POSIX routines, or almost-
--- a/gfx/skia/skia/src/gpu/GrAutoLocaleSetter.h	2018-10-17 21:39:28.000000000 +0000
+++ b/gfx/skia/skia/src/gpu/GrAutoLocaleSetter.h	2018-12-07 17:33:18.878420760 +0000
@@ -25,7 +25,7 @@
 #define HAVE_XLOCALE 0
 #endif
 
-#if defined(SK_BUILD_FOR_ANDROID) || defined(__UCLIBC__) || defined(_NEWLIB_VERSION) || defined(__NetBSD__)
+#if defined(SK_BUILD_FOR_ANDROID) || defined(__UCLIBC__) || defined(_NEWLIB_VERSION) || defined(__NetBSD__) || defined (__sun__)
 #define HAVE_LOCALE_T 0
 #else
 #define HAVE_LOCALE_T 1
--- a/python/mozbuild/mozpack/files.py	2018-10-17 21:39:33.000000000 +0000
+++ b/python/mozbuild/mozpack/files.py	2018-12-07 19:04:10.487895055 +0000
@@ -290,8 +290,8 @@
                 os.path.getsize(self.path) > os.path.getsize(dest):
             return False
         try:
-            if may_strip(dest):
-                strip(dest)
+#            if may_strip(dest):
+#                strip(dest)
             if may_elfhack(dest):
                 elfhack(dest)
         except ErrorMessage:
