--- a/build/moz.configure/init.configure	2018-10-05 01:47:04.000000000 +0000
+++ b/build/moz.configure/init.configure	2020-08-11 18:57:22.468840922 +0000
@@ -940,6 +940,7 @@
 
 
 set_define('XP_SOLARIS', target_is_solaris)
+set_define('SOLARIS', target_is_solaris)
 
 # The application/project to build
 # ==============================================================
--- a/js/src/gc/Memory.cpp	2019-09-03 12:13:28.359033687 -0700
+++ b/js/src/gc/Memory.cpp	2019-09-03 12:13:38.667197092 -0700
@@ -460,7 +460,7 @@
     return true;
 }
 
-bool
+void
 MarkPagesInUse(void* p, size_t size)
 {
     if (!DecommitEnabled())
