--- folks-0.15.3/backends/dummy/lib/meson.build.~1~	2022-01-22 11:01:35.292290937 +0000
+++ folks-0.15.3/backends/dummy/lib/meson.build	2022-01-22 11:02:05.465442255 +0000
@@ -29,7 +29,6 @@
 
 dummy_backendlib_symbolmap = meson.current_source_dir() / 'folks-@0@.map'.format(dummy_backend_name)
 dummy_backendlib_link_flags = cc.get_supported_link_arguments(
-  '-Wl,--version-script,@0@'.format(dummy_backendlib_symbolmap),
 )
 
 dummy_backendlib = shared_library('folks-@0@'.format(dummy_backend_name),
--- folks-0.15.3/backends/telepathy/lib/meson.build.~1~	2022-01-22 11:02:57.993314568 +0000
+++ folks-0.15.3/backends/telepathy/lib/meson.build	2022-01-22 11:03:34.024985262 +0000
@@ -122,7 +122,6 @@
 
 telepathy_backendlib_symbolmap = meson.current_source_dir() / 'folks-@0@.map'.format(telepathy_backend_name)
 telepathy_backendlib_link_flags = cc.get_supported_link_arguments(
-  '-Wl,--version-script,@0@'.format(telepathy_backendlib_symbolmap),
 )
 
 telepathy_backendlib = shared_library('folks-@0@'.format(telepathy_backend_name),
