No pie on 11.3

--- cups-2.4.2/config-scripts/cups-compiler.m4.~1~	2022-07-25 17:23:08.965708278 +0000
+++ cups-2.4.2/config-scripts/cups-compiler.m4	2022-07-25 17:25:04.010575296 +0000
@@ -163,13 +163,6 @@
 		AC_MSG_RESULT([no])
 	    ])
 	], [*], [
-	    CFLAGS="$CFLAGS -fPIE -pie"
-	    AC_COMPILE_IFELSE([AC_LANG_PROGRAM()], [
-		PIEFLAGS="-fPIE -pie"
-		AC_MSG_RESULT([yes])
-	    ], [
-		AC_MSG_RESULT([no])
-	    ])
 	])
 	CFLAGS="$OLDCFLAGS"
     ])
