--- vte-0.64.0/src/modes.py~	2021-04-16 18:22:42.770917700 +0000
+++ vte-0.64.0/src/modes.py	2021-04-17 13:20:16.456782934 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/env python3.7
 #
 # Copyright © 2018, 2020 Christian Persch
 #
--- vte-0.64.0/src/parser-seq.py~	2021-04-16 18:22:42.771917600 +0000
+++ vte-0.64.0/src/parser-seq.py	2021-04-17 13:20:31.364091822 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/env python3.7
 #
 # Copyright © 2015 David Herrmann <dh.herrmann@gmail.com>
 # Copyright © 2018, 2019, 2020 Christian Persch
